Automattic/newspack-custom-content-migrator
# Fix/csv file class

## How to test
Get your most problematic CSV file, and see if this class will read it and process the rows.
```
$ wp shell
> $csv = ( new FileImportFactory )->get_file( '/path/to/file' )->getIterator();
> foreach ($csv as $row ) {
>     var_dump( $row );
> }
```
---

_**I think**_ [this is the best way](https://github.com/Automattic/newspack-custom-content-migrator/blob/fix/csv-file-class/src/Utils/CommonDataFileIterator/CSVFile.php#L131) to handle charsets, but would appreciate if someone more knowledgeable than me could check or confirm.
